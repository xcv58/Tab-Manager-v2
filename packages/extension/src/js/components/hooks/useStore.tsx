import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}

export const useFontSize = () => {
  const { userStore } = useStore()
  return userStore.fontSize
}

export const useTabHeight = () => {
  const fontSize = useFontSize()
  return 3 * fontSize
}
