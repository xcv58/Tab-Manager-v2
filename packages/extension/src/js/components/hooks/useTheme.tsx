import { createContext, useContext } from 'react'

export const ThemeContext = createContext(false)

export const useTheme = () => useContext(ThemeContext)
