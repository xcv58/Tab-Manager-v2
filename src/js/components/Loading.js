import React, { Component } from 'react'

export default class Loading extends Component {
  state = { tooLong: false }

  componentDidMount () {
    this.timer = setTimeout(() => {
      this.setState({ tooLong: true })
      this.timer = null
    }, 100)
  }

  componentWillUnmount () {
    if (this.timer) {
      clearTimeout(this.timer)
      this.timer = null
    }
  }

  render () {
    return (
      <div id='spinner'>
        <div className='la-ball-spin la-dark la-3x'>
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    )
  }
}
