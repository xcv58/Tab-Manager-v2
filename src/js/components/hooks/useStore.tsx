import React from 'react'
import Store from 'stores'

export const store = new Store()

export const StoreContext = React.createContext<Store>(store)

export const useStore = () => {
  return React.useContext(StoreContext)
}
