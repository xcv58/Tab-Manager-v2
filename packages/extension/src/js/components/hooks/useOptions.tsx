import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return null
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  const { historyTabs } = searchStore
  console.log({ historyTabs: [...historyTabs] })
  return [...windowStore.tabs, ...historyTabs]
}
