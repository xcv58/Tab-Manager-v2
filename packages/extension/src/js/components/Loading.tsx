import React from 'react'
import classNames from 'classnames'
import { useTheme } from './hooks/useTheme'

export default ({ small = false }) => {
  const isDarkTheme = useTheme()
  const containerProps = small
    ? { className: 'flex items-center justify-center w-full h-12' }
    : { id: 'spinner' }
  return (
    <div {...containerProps}>
      <div
        className={classNames('la-ball-spin', {
          'la-dark': !isDarkTheme,
          'la-3x': !small,
        })}
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  )
}
