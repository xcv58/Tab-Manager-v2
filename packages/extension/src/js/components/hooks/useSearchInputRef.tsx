import { useRef, useEffect } from 'react'
import { useStore } from 'components/hooks/useStore'

export const useSearchInputRef = () => {
  const searchInputRef = useRef<HTMLInputElement>(null)
  const { searchStore } = useStore()
  const { setSearchEl } = searchStore

  useEffect(() => setSearchEl(searchInputRef))
  return searchInputRef
}
