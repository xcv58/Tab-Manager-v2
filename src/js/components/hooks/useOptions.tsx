import { useStore } from 'components/hooks/useStore'

export const useOptions = () => {
  const { windowStore, shortcutStore, searchStore } = useStore()
  if (searchStore.isCommand) {
    const { shortcuts } = shortcutStore
    return shortcuts
      .map(([shortcut, command, name, hideFromCommand]) => {
        if (typeof name !== 'string' || hideFromCommand) {
          return
        }
        return { name, shortcut, command }
      })
      .filter((x) => x)
      .sort((a, b) => a.name.localeCompare(b.name))
  }
  return windowStore.tabs
}
