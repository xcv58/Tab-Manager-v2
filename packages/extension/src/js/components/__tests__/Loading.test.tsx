import React from 'react'
import { mount } from 'enzyme'
import Loading from 'components/Loading'
import { ThemeContext } from 'components/hooks/useTheme'

describe('Loading should', () => {
  it('render divs', () => {
    const el = mount(<Loading />)
    expect(el.find('div').length).toBe(10)
    const spinner = el.find({ id: 'spinner' })
    expect(spinner.length).toBe(1)
    const la = spinner.find({ className: 'la-ball-spin la-dark la-3x' })
    expect(la.length).toBe(1)
    expect(la.find('div').length).toBe(9)
    expect(el.text()).toBe('')
  })

  it('render divs in dark mode', () => {
    const el = mount(
      <ThemeContext.Provider value>
        <Loading />
      </ThemeContext.Provider>
    )
    expect(el.find('div').length).toBe(10)
    const spinner = el.find({ id: 'spinner' })
    expect(spinner.length).toBe(1)
    const la = el.find({ className: 'la-ball-spin la-3x' })
    expect(la.length).toBe(1)
    expect(la.find('div').length).toBe(9)
    expect(el.text()).toBe('')
  })
})
