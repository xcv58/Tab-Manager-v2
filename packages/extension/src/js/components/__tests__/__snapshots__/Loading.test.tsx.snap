// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render divs in dark mode 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="spinner"
      >
        <div
          class="la-ball-spin la-3x"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="spinner"
    >
      <div
        class="la-ball-spin la-3x"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Loading should render divs with default props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="spinner"
      >
        <div
          class="la-ball-spin la-dark la-3x"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="spinner"
    >
      <div
        class="la-ball-spin la-dark la-3x"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Loading should render small component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center w-full h-12"
      >
        <div
          class="la-ball-spin la-dark"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center w-full h-12"
    >
      <div
        class="la-ball-spin la-dark"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
